src/osd/OSD.cc

```diff
@@ -6706,6 +6719,9 @@ void OSD::handle_osd_map(MOSDMap *m)
   // even if this map isn't from a mon, we may have satisfied our subscription
   monc->sub_got("osdmap", last);
 
+  if (!m->maps.empty())
```

Comment:
I put the logic inside the finish_full_map_request function, so if monitor is not capping the MOSDMap, the requested_full_first = requested_full_last =0, the log will not be printed out.

Or,

Whould you prefer to change the code like 

if ((requested_full_first <= requested_full_last && ! m->maps.empty() )
     finish_full_map_request();

Is this more clear?